python/serve.py

```
@@ -98,29 +227,200 @@ def generate(
+        return max(range(vocab), key=lambda i: logits[i])
+
+    # Build (index, logit) pairs
+    indexed = list(enumerate(logits[:vocab]))
+
+    # Top-k filtering
+    if top_k > 0:
+        indexed.sort(key=lambda x: x[1], reverse=True)
+        indexed = indexed[:top_k]
+
+    # Temperature-scaled softmax
+    max_l = max(l for _, l in indexed)
```

Ruff (0.15.1)

[error] Ambiguous variable name: `l`

(E741)